pycbc/inference/sampler/base_mcmc.py

```diff
@@ -464,14 +486,6 @@ def run(self):
         # "nsamples" keeps track of the number of samples we've obtained (if
         # target_eff_nsamples is not None, this is the effective number of
         # samples; otherwise, this is the total number of samples).
-        # _lastclear is the number of iterations that the file already
-        # contains (either due to sampler burn-in, or a previous checkpoint)
-        if self.new_checkpoint:
-            self._lastclear = 0
-        else:
-            with self.io(self.checkpoint_file, "r") as fp:
-                self._lastclear = fp.niterations
-                self.thin_interval = fp.thinned_by
```

Comment:
Put this block of code back, as it's needed to make sure the thinning is done correctly.